src/main/java/command/AddTaskCommand.java

```diff
@@ -0,0 +1,33 @@
+package command;
+
+import core.Common;
+import core.Ui;
+import dukexception.DukeException;
+import task.Task;
+
+/**
+ * Generic command to add a new task.
+ */
+public abstract class AddTaskCommand extends Command{
+
+    private Task task;
+
+    /**
+     * Constructor to create a new task.
+     * @param task to be added to the storage.
+     */
+    public AddTaskCommand(Task task){
+        this.task=task;
+    }
+
+    /**
+     * Executes to add the new task to the storage.
+     * @param common to add task to the data storage.
+     * @param ui to display the response text of adding task.
+     * @throws DukeException when adding of the new task is unsuccessful.
+     */
+    @Override
+    public void execute(Common common, Ui ui) throws DukeException {
```

Comment:
It would be better not to include ui in the execute command, to reduce coupling in the methods. Maybe you could return true or false if command is successful, then ui can decide what to do with the result.